cpp/unreal_plugins/SpCore/Source/SpCore/Unreal.h

```diff
@@ -212,40 +212,61 @@ class SPCORE_API Unreal
         std::derived_from<TComponent, TReturnAsComponent>
     static TReturnAsComponent* createComponentInsideOwnerConstructor(AActor* owner, const std::string& component_name)
     {
-        SP_ASSERT(owner);
-        TReturnAsComponent* actor_component = owner->CreateDefaultSubobject<TComponent>(toFName(component_name));
-        SP_ASSERT(actor_component);
-        return actor_component;
+        return static_cast<TReturnAsComponent*>(createComponentInsideOwnerConstructorByClass(TComponent::StaticClass(), owner, component_name));
```

Comment:
Use `Cast` instead of `static_cast` as an extra safety precaution, although strictly speaking, the surrounding logic in this case should guarantee that `static_cast` should be fine too.